src/abstractdataframe/abstractdataframe.jl

```diff
@@ -84,6 +88,7 @@ end
 
 Base.names(df::AbstractDataFrame, T::Type) =
     [String(n) for (n, c) in pairs(eachcol(df)) if eltype(c) <: T]
+Base.names(df::AbstractDataFrame, fun::Function) = filter(fun, names(df))
```

Comment:
IIRC `names` always return a copy so `filter!` is safe, right?
```suggestion
Base.names(df::AbstractDataFrame, fun::Function) = filter!(fun, names(df))
```